js-core/homeworks/homework-15/src/main.js

```diff
@@ -0,0 +1,386 @@
+        this.setClass(input, 'form-control');
+        
+        this.setId(input, 'search');
+        this.setTypeForInput(input, 'text');
+        this.setPlaceHolderForInput(input, 'search');
+        this.setLableFor(label, 'search');
+        
+        formGroup.appendChild(label);
+        formGroup.appendChild(input);
+        
+        return formGroup;
+    },
+    
+    fillTableContactList() {
+        /*Create thead element and fill it*/
+        const thead = this.newElem('thead');
+        const trForThead = this.newElem('tr');
+        
+        contentForThead.forEach(contentElem => {
+            const th = this.newElem('th');
+            this.setTextContent(th, contentElem);
+            trForThead.appendChild(th);
+        });
+        
+        thead.appendChild(trForThead);
+        
+        /*Create tbody element and fill it*/
+        const tbody = this.newElem('tbody');
+        
+        users.forEach(user => {
+            const tr = this.newElem('tr');
+            
+            user.forEach(userInfo => {
+                const name = this.newElem('td');
+                const lastName = this.newElem('td');
+                const email = this.newElem('td');
+                
+                this.setTextContent(name, userInfo.name);
+                this.setTextContent(lastName, userInfo.lastName);
+                this.setTextContent(email, userInfo.email);
+                
+                tr.appendChild(name);
+                tr.appendChild(lastName);
+                tr.appendChild(email);
+            });
+            tbody.appendChild(tr);
+        });
+        
+        /*Append these element to table*/
+        this.table.appendChild(thead);
+        this.table.appendChild(tbody);
+    },
+    
+    createMainContent() {
+        this.main = this.newElem('main');
+        const conteiner = this.newElem('div');
+        this.setClass(conteiner, 'container');
+        
+        /*There is implementation of search form*/
+        const searchForm = this.newElem('form');
+        this.setClass(searchForm, 'form-inline search-form');
+        searchForm.appendChild(this.fillSearchForm()); // fillSearchForm is above
+        
+        /*There is implementation of contact list (table)*/
+        this.table = this.newElem('table');
+        this.setClass(this.table, 'table table-hover contacts');
+        this.fillTableContactList();
+        
+        conteiner.appendChild(searchForm);
+        conteiner.appendChild(this.table);
+        this.main.appendChild(conteiner);
+    },
+    
+    fillFooter() {
+        const nav = this.newElem('nav');
+        this.setClass(nav, 'main-nav');
+        
+        footerTabs.forEach((tab, index) => {
+            const link = this.newElem('a');
+            const icon = this.newElem('span');
+            const tabContent = this.newElem('span');
+            
+            this.setClass(link, 'tab');
+            this.setClass(icon, `glyphicon glyphicon-${idOfIcons[index]}`);
```

Comment:
it's not good to have a dependency to index

For future probably in such case, you should rename idOfIcons -> sortedIdOfIcons because of they depend on order